libs/@local/codec/src/numeric/real.rs

```diff
@@ -530,6 +612,25 @@ mod tests {
         let large_value = Real::try_from(f64::MAX).expect("Failed to convert f64::MAX to Real");
 
         // Should fail to give an accurate result
-        assert!(large_value.to_f32().is_infinite());
+        assert!(large_value.to_f32_lossy().is_infinite());
+    }
+
+    #[test]
+    #[expect(clippy::float_cmp, clippy::decimal_literal_representation)]
+    fn lossy_vs_exact_conversion() {
+        let inside_f32 = Real::from_natural(123_456, 0); // 123456 - needs ~17 bits
+        let inside_f64 = Real::from_natural(16_777_217, 0); // 2^24 + 1 - needs 25 bits
+        let outside_f64 = Real::from_natural(123_456_789, 20); // 123456789 * 10^20
```

Comment:
Can't we just pass the exponent? e.g. `from_natural(1, 17)`? That's more intuitive.

Reply:
in theory yes, but the issue described only happens if the significand is larger than eg 53 bits, which wouldn't happen if we use an exponent because the exponent is saved separately